internal/tableland/impl/mesa_instrumented.go

```diff
@@ -98,28 +91,29 @@ func (t *InstrumentedTablelandMesa) RelayWriteQuery(
 // GetReceipt returns the receipt for a txn hash.
 func (t *InstrumentedTablelandMesa) GetReceipt(
 	ctx context.Context,
-	req tableland.GetReceiptRequest,
-) (tableland.GetReceiptResponse, error) {
+	txnHash string,
+) (bool, *tableland.TxnReceipt, error) {
 	start := time.Now()
-	resp, err := t.tableland.GetReceipt(ctx, req)
+	ok, resp, err := t.tableland.GetReceipt(ctx, txnHash)
 	latency := time.Since(start).Milliseconds()
 
 	chainID, _ := ctx.Value(middlewares.ContextKeyChainID).(tableland.ChainID)
 	t.record(ctx, recordData{"GetReceipt", "", "", err == nil, latency, chainID})
-	return resp, err
+	return ok, resp, err
 }
 
 // SetController allows users to the controller for a token id.
-func (t *InstrumentedTablelandMesa) SetController(ctx context.Context,
-	req tableland.SetControllerRequest,
-) (tableland.SetControllerResponse, error) {
+func (t *InstrumentedTablelandMesa) SetController(
+	ctx context.Context,
+	controller common.Address,
+	tableID tables.TableID,
+) (tables.Transaction, error) {
 	start := time.Now()
-	resp, err := t.tableland.SetController(ctx, req)
+	resp, err := t.tableland.SetController(ctx, controller, tableID)
 	latency := time.Since(start).Milliseconds()
 
-	controller, _ := ctx.Value(middlewares.ContextKeyAddress).(string)
 	chainID, _ := ctx.Value(middlewares.ContextKeyChainID).(tableland.ChainID)
-	t.record(ctx, recordData{"SetController", controller, "", err == nil, latency, chainID})
+	t.record(ctx, recordData{"SetController", controller.Hex(), "", err == nil, latency, chainID})
```

Comment:
I think there was a little bug here. In the old code, it was pulling an address out of the context using `middlewares.ContextKeyAddress` then using that address to record as the `controller`. But the address in the context is actually the `caller`. The new code just uses the provided `controller` param to record the controller.

Reply:
I think it kind of depends on if we want to record who made the call versus which is the target new controller. If I'd try to make the interpretation today, I think I'd lean to what you did but would still have doubts.

In any case, we should really remove that `controller` recording in metrics. That leads to high cardinality dimensions that aren't nice for Prometheus when we have more volume. So probably in the medium term, it will be removed, so the dilemma will be solved. ha

Reply:
Ok will leave as is.